packages/aws-cdk-lib/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -460,11 +507,17 @@ export class StateMachine extends StateMachineBase {
       tracingConfiguration: props.tracingEnabled ? this.buildTracingConfiguration() : undefined,
       ...definitionBody.bind(this, this.role, props, graph),
       definitionSubstitutions: props.definitionSubstitutions,
+      encryptionConfiguration: props.kmsKey? {
+        kmsKeyId: props.kmsKey.keyArn,
+        kmsDataKeyReusePeriodSeconds: props.kmsDataKeyReusePeriodSeconds? props.kmsDataKeyReusePeriodSeconds.toSeconds() : 300,
```

Comment:
Extract 300 to constant variable. How did you determine this to be a sensible default?

Reply:
If you don't provide `kmsDataKeyReusePeriodSeconds` the service will default to 300.

> KmsDataKeyReusePeriodSeconds - Maximum duration for which SFN will reuse data keys. When the period expires, Step Functions will call GenerateDataKey. This setting can only be set when Type is CUSTOMER_MANAGED_KMS_KEY. The value can range from 60-900 seconds. Default is 300 seconds.

https://docs.aws.amazon.com/step-functions/latest/dg/encryption-at-rest.html#cfn-resources-for-encryption-configuration